decidim-initiatives/app/commands/decidim/initiatives/vote_initiative.rb

```diff
@@ -94,8 +94,9 @@ def notify_percentage_change(before, after)
         )
       end
 
-      def notify_support_threshold_reached(percentage)
-        return unless percentage >= 100
+      def notify_support_threshold_reached(before, after)
+        # Don't need to notify if threshold has already been reached
+        return if before == after || after != 100
```

Comment:
The notification only has to be sent the first time threshold is reached?
Maybe this condition could be as:
return unless (after >= 100 && before <= 100)  

If before/after values calculated in basis a vote (+1), the calculated before/after percentages wouldn't be always different ?
And the second condition, "after != 100", includes those cases where percentage would be before reaching 100 but after having reached it too, so, only when after == 100 could pass it ?

Reply:
Hi,

No, the percentage cannot be above 100%, therefore:
- if after is not 100%, that's mean that there is not enough vote.
- if before == after, that's mean that the threshold is already achieved

in your proposition, an initiative that has already reached the threshold may send a notification again.

We could refactor it to 
`return if before >= 100 || after != 100`

What do you think about it?

Reply:
Oh, ok @armandfardeau I didnt saw that fact, sorry for that!! :pray: 
Then if percentage cannot be above 100%, then I see ok your first aproximation, since before cannot be above 100, there's no sense to check if it can be > it on first condition.